File: projects/16ffbf18/generate-icons.js
// Generate PWA icons using simple base64 PNG data
const fs = require('fs');
const path = require('path');

// Create icons directory if it doesn't exist
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
    fs.mkdirSync(iconsDir);
}

// Simple function to create a colored square PNG with text
function generateIcon(size, filename) {
    // This is a very basic approach - creating a simple red and black icon
    // In production, you'd want to use a proper image library like 'sharp' or 'canvas'

    const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="${size}" height="${size}">
    <rect width="512" height="512" fill="#000000"/>
    <circle cx="256" cy="200" r="80" fill="#dc3545"/>
    <rect x="176" y="280" width="160" height="40" rx="20" fill="#dc3545"/>
    <rect x="140" y="340" width="232" height="40" rx="20" fill="#ffffff"/>
    <rect x="140" y="400" width="232" height="40" rx="20" fill="#ffffff"/>
</svg>`;

    const svgPath = path.join(iconsDir, filename.replace('.png', '.svg'));
    fs.writeFileSync(svgPath, svg);

    console.log(`Created ${filename} (SVG format)`);

    // Note: For actual PNG conversion, you would need a library like 'sharp' or use ImageMagick
    // For now, we'll create SVG versions which modern browsers can use
}

// Generate all required sizes
const sizes = [
    { size: 72, name: 'icon-72x72.png' },
    { size: 96, name: 'icon-96x96.png' },
    { size: 128, name: 'icon-128x128.png' },
    { size: 144, name: 'icon-144x144.png' },
    { size: 152, name: 'icon-152x152.png' },
    { size: 192, name: 'icon-192x192.png' },
    { size: 384, name: 'icon-384x384.png' },
    { size: 512, name: 'icon-512x512.png' }
];

sizes.forEach(({ size, name }) => {
    generateIcon(size, name);
});

console.log('Icon generation complete!');
console.log('Note: SVG files created. For PNG conversion, use ImageMagick or a similar tool.');

File: projects/16ffbf18/generate-icons.py
#!/usr/bin/env python3
"""Generate PWA icons with red and black theme"""

import os
from PIL import Image, ImageDraw

# Create icons directory
os.makedirs('icons', exist_ok=True)

def generate_icon(size, filename):
    """Generate a single icon with the specified size"""
    # Create a black background
    img = Image.new('RGB', (size, size), color='#000000')
    draw = ImageDraw.Draw(img)

    # Calculate scaled dimensions
    scale = size / 512

    # Draw red circle (info icon)
    circle_center = (size // 2, int(size * 0.39))
    circle_radius = int(size * 0.156)
    draw.ellipse(
        [
            circle_center[0] - circle_radius,
            circle_center[1] - circle_radius,
            circle_center[0] + circle_radius,
            circle_center[1] + circle_radius
        ],
        fill='#dc3545'
    )

    # Draw red rounded rectangle (top element)
    rect1_x = int(size * 0.344)
    rect1_y = int(size * 0.547)
    rect1_w = int(size * 0.313)
    rect1_h = int(size * 0.078)
    draw.rounded_rectangle(
        [rect1_x, rect1_y, rect1_x + rect1_w, rect1_y + rect1_h],
        radius=int(size * 0.039),
        fill='#dc3545'
    )

    # Draw white rounded rectangles (info lines)
    rect2_x = int(size * 0.273)
    rect2_y = int(size * 0.664)
    rect2_w = int(size * 0.453)
    rect2_h = int(size * 0.078)
    draw.rounded_rectangle(
        [rect2_x, rect2_y, rect2_x + rect2_w, rect2_y + rect2_h],
        radius=int(size * 0.039),
        fill='#ffffff'
    )

    rect3_x = int(size * 0.273)
    rect3_y = int(size * 0.781)
    rect3_w = int(size * 0.453)
    rect3_h = int(size * 0.078)
    draw.rounded_rectangle(
        [rect3_x, rect3_y, rect3_x + rect3_w, rect3_y + rect3_h],
        radius=int(size * 0.039),
        fill='#ffffff'
    )

    # Save the image
    filepath = os.path.join('icons', filename)
    img.save(filepath, 'PNG')
    print(f'Created {filename}')

# Generate all required sizes
sizes = [
    (72, 'icon-72x72.png'),
    (96, 'icon-96x96.png'),
    (128, 'icon-128x128.png'),
    (144, 'icon-144x144.png'),
    (152, 'icon-152x152.png'),
    (192, 'icon-192x192.png'),
    (384, 'icon-384x384.png'),
    (512, 'icon-512x512.png')
]

for size, filename in sizes:
    generate_icon(size, filename)

print('Icon generation complete!')
